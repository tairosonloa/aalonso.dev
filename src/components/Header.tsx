/* eslint-disable jsx-a11y/anchor-is-valid */
import { Transition } from '@headlessui/react'
import Link from 'next/link'
import React, { FC, useState } from 'react'

const Header: FC = () => {
  const [isOpen, setIsOpen] = useState(false)
  const navMenu = [
    { text: 'Home', page: '/' },
    { text: 'Blog', page: '/blog' },
    { text: 'About', page: '/about' },
  ]

  return (
    <header className="fixed top-0 w-full z-20">
      <nav className="border-b border-gray-800">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <div className="flex items-center justify-between h-16">
            <div className="flex items-center justify-between w-full">
              <div
                className="flex-shrink-0 cursor-pointer"
                onKeyPress={() => setIsOpen(false)}
                onClick={() => setIsOpen(false)}
                role="button"
                tabIndex={0}
              >
                <Link href="/">
                  <a aria-label="Logo" className="text-3xl md:text-4xl font-bold">
                    aalonso
                  </a>
                </Link>
              </div>
              <div className="hidden md:block">
                <div className="ml-10 flex items-baseline space-x-4">
                  {navMenu.map((item) => (
                    <Link href={item.page} key={item.text}>
                      <a className="hover:bg-gray-800 focus:bg-gray-800 px-3 py-2 rounded-md text-sm lg:text-base font-medium">
                        {item.text}
                      </a>
                    </Link>
                  ))}
                </div>
              </div>
            </div>

            {/* <!-- Mobile menu button --> */}
            <div className="-mr-2 flex md:hidden">
              <button
                onClick={() => setIsOpen(!isOpen)}
                type="button"
                className="bg-gray-800 inline-flex items-center justify-center p-2 rounded-md text-gray-300 hover:text-gray-100 hover:bg-gray-800 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-transparent"
                aria-controls="mobile-menu"
                aria-expanded="false"
              >
                <span className="sr-only">Open main menu</span>
                <svg
                  className="block h-6 w-6"
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke="currentColor"
                  aria-hidden="true"
                >
                  {!isOpen ? (
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      d="M4 6h16M4 12h16M4 18h16"
                    />
                  ) : (
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      d="M6 18L18 6M6 6l12 12"
                    />
                  )}
                </svg>
              </button>
            </div>
          </div>
        </div>

        {/* <!-- Mobile menu --> */}
        <Transition
          show={isOpen}
          enter="linear duration-100 transform"
          enterFrom="opacity-0 scale-95"
          enterTo="opacity-100 scale-100"
          leave="linear duration-100 transform"
          leaveFrom="opacity-100 scale-100"
          leaveTo="opacity-0 scale-95"
        >
          {() => (
            <div className="h-screen flex flex-col items-center pt-14 md:hidden" id="mobile-menu">
              {navMenu.map((item) => (
                <div onClick={() => setIsOpen(!isOpen)} aria-hidden="true" key={item.text}>
                  <Link href={item.page}>
                    <a className="hover:bg-gray-800 focus:bg-gray-800 block px-3 py-2 rounded-md text-xl font-medium mb-6">
                      {item.text}
                    </a>
                  </Link>
                </div>
              ))}
            </div>
          )}
        </Transition>
      </nav>
    </header>
  )
}

export default Header
