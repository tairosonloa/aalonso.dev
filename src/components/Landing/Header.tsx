/* eslint-disable jsx-a11y/accessible-emoji */
/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react'

const Main = () => {
  return (
    <main className="flex flex-col space-y-12 divide-y divide-gray-200 mb-32 ">
      <div>
        <h1 className="text-4xl md:text-5xl lg:text-6xl font-extrabold leading-tight tracking-tight text-gray-100  sm:leading-10   md:leading-14">
          Hi, I&apos;m PranavðŸ‘‹
        </h1>
        <p className="mt-3 mb-6 md:mb-7 sm:w-9/12 text-gray-300 md:w-10/12 lg:w-full text-base md:text-lg font-medium ">
          I am a software developer building beautiful interfaces and accessible applications.
        </p>

        {/* <button
          type="button"
          className="text-sm font-bold lg:text-base tracking-normal my-5 md:my-7 p-2.5 px-4 md:px-5 rounded ring-2 ring-gray-400 ring-opacity-50 bg-gray-900 bg-opacity-50 hover:bg-opacity-100 focus:bg-opacity-100 shadow transition"
        >
          Resume
        </button> */}

        <a
          href="mailto:prnvbirajdar@gmail.com"
          className="mt-6 text-sm font-bold lg:text-base tracking-normal my-5 md:my-7 p-2.5 px-4 md:px-5 rounded ring-2 ring-gray-400 ring-opacity-50 bg-gray-900 bg-opacity-50 hover:bg-opacity-100 focus:bg-opacity-100 shadow transition"
        >
          Say Hello!
        </a>
      </div>
    </main>
  )
}

export default Main
