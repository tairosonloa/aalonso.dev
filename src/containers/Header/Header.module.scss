@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.container {
  position: fixed;
  z-index: 20;
  top: 0;
  left: 0;
  width: 100%;
}

.contentContainer {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: grid_units(2) grid_units(5);
}

.titleContainer {
  min-height: grid_units(11);
}

.title {
  position: sticky;
  margin-bottom: grid_units(2);

  @include screenIn(smallTablet) {
    margin-top: grid_units(1);
    margin-bottom: grid_units(1);
  }

  font-size: 2.25rem;
  font-weight: bold;
  text-transform: none;

  em {
    color: $burgundy;
    font-style: normal;
  }
}
.menu {
  border-color: $grey-light;
  color: $grey-light;

  @include screenIn(smallTablet) {
    margin-top: grid_units(3);
  }

  * {
    border-color: $grey-light;
    color: $grey-light;
  }
}

.solid {
  border-color: $black;
  background-color: $white;
  color: $black;

  * {
    border-color: $black;
    color: $black;
  }
}

.shadow {
  box-shadow: 0 0 20px -10px $rich-black;
}
