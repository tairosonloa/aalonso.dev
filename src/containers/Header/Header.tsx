import classNames from 'classnames'
import React, { FC, useState } from 'react'
import { Typography, TypographyType } from '../../components/DataDisplays/Typography/Typography'
import { Menu } from '../../components/Menu/Menu'
import { useScrollPosition } from '../../hooks/useScrollPosition'
import { Router } from '../../services/router/Router'
import { Route } from '../../services/router/Routes'
import styles from './Header.module.scss'

type HeaderProps = {
  links: { href: string | Route; text: string }[]
  router: Router
  fixed?: boolean
}

export const Header: FC<HeaderProps> = ({ links, router, fixed = false }) => {
  const [solid, setSolid] = useState<boolean>(fixed)
  const [shadow, setShadow] = useState<boolean>(false)
  const HEADER_HEIGHT = 75

  useScrollPosition(
    ({ currentPosition }) => {
      if (!fixed) setSolid(currentPosition.y > window.innerHeight - HEADER_HEIGHT)
      setShadow(currentPosition.y > HEADER_HEIGHT)
    },
    [solid, shadow],
  )

  return (
    <header
      className={classNames(
        styles.container,
        solid && styles.solid,
        solid && shadow && styles.shadow,
      )}
    >
      <div className={styles.contentContainer}>
        <div className={styles.titleContainer}>
          {solid && (
            <Typography typographyType={TypographyType.TITLE_BIG} className={styles.title}>
              aa<em>l</em>onso
            </Typography>
          )}
        </div>
        <Menu links={links} router={router} className={styles.menu} />
      </div>
    </header>
  )
}

export default Header
