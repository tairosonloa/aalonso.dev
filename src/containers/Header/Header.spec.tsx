import { render } from '@testing-library/react'
import * as React from 'react'
import { Header } from './Header'

const headerLinks = [
  { href: '#bio', text: 'About me' },
  { href: '#skills', text: 'Skills' },
  { href: '#career', text: 'Career' },
]

describe('<Header/>', () => {
  it.each(headerLinks)('should display the links to $href with text $text', ({ href, text }) => {
    const renderResult = render(<Header links={headerLinks} />)
    expect(renderResult.getByText(text)).toBeTruthy()
    expect(renderResult.getByTestId(text)).toHaveAttribute('href', href)
  })
})
