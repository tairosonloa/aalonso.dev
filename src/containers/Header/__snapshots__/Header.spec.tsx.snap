// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: unset;"
  >
    <div>
      <header
        class="container"
      >
        <div
          class="contentContainer"
        >
          <div
            class="titleContainer"
          />
          <div
            class="container menu"
          >
            <button
              aria-label="Menu"
              class="button"
              type="button"
            >
              <i
                class="ri-menu-line icon buttonIcon activeIcon"
              />
              <i
                class="ri-close-line icon buttonIcon closeIcon"
              />
            </button>
            <nav
              class="links"
            >
              <a
                data-testid="About me"
                href="#bio"
              >
                <h2
                  class="common titleSmall"
                >
                  About me
                </h2>
              </a>
              <a
                data-testid="Skills"
                href="#skills"
              >
                <h2
                  class="common titleSmall"
                >
                  Skills
                </h2>
              </a>
              <a
                data-testid="Career"
                href="#career"
              >
                <h2
                  class="common titleSmall"
                >
                  Career
                </h2>
              </a>
            </nav>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="container"
    >
      <div
        class="contentContainer"
      >
        <div
          class="titleContainer"
        />
        <div
          class="container menu"
        >
          <button
            aria-label="Menu"
            class="button"
            type="button"
          >
            <i
              class="ri-menu-line icon buttonIcon activeIcon"
            />
            <i
              class="ri-close-line icon buttonIcon closeIcon"
            />
          </button>
          <nav
            class="links"
          >
            <a
              data-testid="About me"
              href="#bio"
            >
              <h2
                class="common titleSmall"
              >
                About me
              </h2>
            </a>
            <a
              data-testid="Skills"
              href="#skills"
            >
              <h2
                class="common titleSmall"
              >
                Skills
              </h2>
            </a>
            <a
              data-testid="Career"
              href="#career"
            >
              <h2
                class="common titleSmall"
              >
                Career
              </h2>
            </a>
          </nav>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
